dotnet/src/Microsoft.AutoGen/RuntimeGateway.Grpc/Services/Grpc/GrpcGatewayService.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// GrpcGatewayService.cs
+using Grpc.Core;
+using Microsoft.AutoGen.Protobuf;
+
+namespace Microsoft.AutoGen.RuntimeGateway.Grpc;
+
+/// <summary>
+/// Represents the gRPC service which handles communication between the agent worker and the cluster.
+/// </summary>
+public sealed class GrpcGatewayService(GrpcGateway gateway) : AgentRpc.AgentRpcBase
+{
+    private readonly GrpcGateway Gateway = (GrpcGateway)gateway;
+
+    /// <summary>
+    /// Method run on first connect from a worker process.
+    /// </summary>
+    /// <param name="requestStream">The request stream.</param>
+    /// <param name="responseStream">The response stream.</param>
+    /// <param name="context">The server call context.</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public override async Task OpenChannel(IAsyncStreamReader<Message> requestStream, IServerStreamWriter<Message> responseStream, ServerCallContext context)
```

Comment:
It looks like the control channel is not handled in this PR

Reply:
its not - we can add it